DEVELOPER.md

```diff
@@ -26,8 +26,42 @@ Envoy Gateway is built using a [make][make]-based build system. Our CI is based
   separately.
 
 ## Quick start
+* Run `make help` to see all the available targets to build, test and run `envoy-gateway`.
```

Comment:
@arkodg one last nit, do you mind reviewing all the bullets?

Reply:
this page already has bullets, prefer keeping the styling the same, similar to the rest of the docs such as [design](https://github.com/envoyproxy/gateway/blob/main/docs/design/SYSTEM_DESIGN.md) docs in this repo